mapic/mistapiconnector_app.go

```diff
@@ -423,38 +445,107 @@ func (mc *mac) shouldEnableAudio(stream *api.Stream) bool {
 	return audio
 }
 
-func (mc *mac) startMultistream(wildcardPlaybackID, playbackID string, info *streamInfo) {
-	for i := range info.stream.Multistream.Targets {
-		go func(targetRef api.MultistreamTargetRef) {
-			glog.Infof("==> starting multistream %s", targetRef.ID)
-			target, pushURL, err := mc.getPushUrl(info.stream, &targetRef)
-			if err != nil {
-				glog.Errorf("Error building multistream target push URL. targetId=%s stream=%s err=%v",
-					targetRef.ID, wildcardPlaybackID, err)
-				return
-			} else if target.Disabled {
-				glog.Infof("Ignoring disabled multistream target. targetId=%s stream=%s",
-					targetRef.ID, wildcardPlaybackID)
-				return
+func (mc *mac) reconcileMultistreamLoop(ctx context.Context) {
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+		case <-mc.multistreamUpdated:
+		}
+		mc.reconcileMultistream()
+	}
+}
+
+func (mc *mac) reconcileMultistream() {
```

Comment:
Similarly here, the PR description is great but it would be good to get some commentary into the code about how we're reconciling the streams vs. Mist's records of them, why we need to call this on a loop etc.

Reply:
Updated in 351b6385c825aceb2cf33c6897e8e88a3ee84076